game/game/cards.lumber

```diff
@@ -0,0 +1,19 @@
+:- use(@core).
+:- use(@core::list).
+
+:- pub(takeCard/3).
+takeCard(G, Id, G2) :-
+    G =:= game { cards: Cards, ..State },
+    remove(Cards, card { id: Id, .. }, Cards2),
+    G2 =:= game { cards: Cards2, ..State }.
+
+:- pub(card/3).
+card(G, Id, Kind) :-
+  G =:= game { cards: Cards, discards: Discards, .. },
+  Card =:= card { id: Id, card: card(Kind), .. },
+  in(Card, Cards),
+  notin(Card, Discards).
+
+:- pub(cardSuit/3).
+cardSuit(game { cards: Cards, ..}, Id, Suit ) :-
+  in(card { id: Id, suit: !suit(Suit), .. }, Cards).
```

Comment:
Are these being used anywhere? I think not? I think they can be the same function anyway, which just checks the details of a card by ID, not worrying about who owns it or if its discarded. `isDiscarded/2` can be a separate one if needed.

```
card(G, Id, Kind, Suit)
```